src/cls/rgw/cls_rgw_ops.h

```diff
@@ -1030,21 +1030,26 @@ struct cls_rgw_lc_get_next_entry_op {
 };
 WRITE_CLASS_ENCODER(cls_rgw_lc_get_next_entry_op)
 
-using rgw_lc_entry_t = std::pair<std::string, int>;
-
 struct cls_rgw_lc_get_next_entry_ret {
-  rgw_lc_entry_t entry;
+  cls_rgw_lc_entry entry;
+
   cls_rgw_lc_get_next_entry_ret() {}
 
   void encode(bufferlist& bl) const {
-    ENCODE_START(1, 1, bl);
+    ENCODE_START(2, 2, bl);
     encode(entry, bl);
     ENCODE_FINISH(bl);
   }
 
   void decode(bufferlist::const_iterator& bl) {
-    DECODE_START(1, bl);
-    decode(entry, bl);
+    DECODE_START(2, bl);
+    if (struct_v < 1) {
+      std::pair<std::string, int> oe;
+      decode(oe, bl);
+      entry = {oe.first, 0 /* start */, uint32_t(oe.second)};
+    } else {
+      decode(entry, bl);
+    }
```

Comment:
@mattbenjamin this should be `if (struct_v <= 1)` right? otherwise we'll try to decode version 1 as `cls_rgw_lc_entry` when it was encoded as `pair<string,int>`. same with cls_rgw_lc_rm_entry_op and cls_rgw_lc_set_entry_op

Reply:
yes, that must be the intent;  that said, the case arises only on an un-upgraded osd, which I think explains why it wasn't caught

Reply:
this part is tracked in https://tracker.ceph.com/issues/46838

Reply:
@cbodley created a tracker issue for this on master, and I think it would be safe to take that fix as a backport